packages/support/docs/03-icons.md

```diff
@@ -33,7 +33,7 @@ use Filament\Support\Facades\FilamentIcon;
 
 FilamentIcon::register([
     'panels::topbar.global-search.field' => 'fas-magnifying-glass',
-    'panels::sidebar.group.collapse-button' => view('icons.chevron-up'),
+    'panels::sidebar.group.collapse-button' => new HtmlString(view('icons.chevron-up')->render()),
```

Comment:
Shouldn't passing a view just work? Maybe it needs to be `fn () => view()` instead?

Reply:
Not with the current solution. A view gets rendered as a string when passed as an attribute and isn't of type `Htmlable` anymore. I might try the slot solution.

Reply:
Just to clarify and to have my comment on Discord here too.

A view gets rendered to a string because it implements the magic method `__toString()` and `HasIcon` did not accept `Htmlable`. PHP converts it to a string to match the `string` type on properties and methods.

The reason it doesn't work when you also accept `Htmlable` is because views also implement `Renderable`. All data passed to views an its derivatives (e.g. Blade components) will get processed by [`gatherData`](https://github.com/laravel/framework/blob/11.x/src/Illuminate/View/View.php#L221) and rendered to HTML.

As a solution/workaround I have added the `Htmlable` type to the `HasIcon` trait and included the following in `getIcon` to make sure we're not passing a Renderable to views.

```php
$icon = $this->icon instanceof Renderable
    ? new HtmlString($this->icon->render())
    : $this->icon;
```

`HtmlString` is the simplest solution in this case, but type hints should still be `Htmlable` as it's the only implementation you need for HTML content in Laravel.